File: OrdersMicroservice/Models/Database/Order.cs

namespace OrdersMicroservice.Models.Database
{
    public class Order
    {
        public string Id { get; set; }

        public string UniqueId { get; set; } = string.Empty;

        public string OrderSum { get; set; } = string.Empty;

        public string Comment { get; set; } = string.Empty;

        public OrderStatusEnum Status { get; set; }

        public List<OrderItem> OrderItems { get; set; } = default!;

        public Address Address { get; set; } = default!;

        public Order()
        {
            Id = Guid.NewGuid().ToString();
        }
    }
}

File: client/src/utilities/models/account/Order.ts
import Address from "./Address";
import { FullOrderItem } from "./OrderItem";

export interface Order {
  id: string;
  uniqueId: string;
  orderSum: string;
  status: OrderStatusEnum;
  orderDate: Date;
}

export interface OrderWithItems extends Order {
  orderItems: FullOrderItem[];
  address: Address;
}

export enum OrderStatusEnum {
  New,
  Approved,
  Processed,
  Shipped,
  Delivered,
  Cancelled,
}

export const statuses = [
  { key: "New", id: 0 },
  { key: "Approved", id: 1 },
  { key: "Processed", id: 2 },
  { key: "Shipped", id: 3 },
  { key: "Delivered", id: 4 },
  { key: "Cancelled", id: 5 },
];
